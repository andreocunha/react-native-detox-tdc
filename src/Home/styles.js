import { StyleSheet } from 'react-native';


const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: '#fff',
    },
    title: {
      fontSize: 30,
      fontWeight: 'bold',
      marginBottom: 40,
      color: '#000',
    },
    input: {
      width: '80%',
      height: 44,
      padding: 8,
      borderRadius: 8,
      borderWidth: 1,
      borderColor: '#000',
      marginBottom: 10,
      fontSize: 18,
    }
});

export default styles;