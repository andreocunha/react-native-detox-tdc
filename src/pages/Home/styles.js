import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: '#fff',
    },
    title: {
      fontSize: 30,
      fontWeight: 'bold',
      marginBottom: 10,
      color: '#051933',
    },
    loginArea: {
      width: '100%',
      padding: 30,
    },
    input: {
      width: '100%',
      height: 44,
      padding: 8,
      borderRadius: 8,
      borderWidth: 1,
      borderColor: '#000',
      marginBottom: 20,
      fontSize: 18,
    },
    button: {
      backgroundColor: '#051933',
      width: 100,
      height: 44,
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: 8,
    },
    textButton: {
      color: '#fff',
      fontSize: 16,
      fontWeight: 'bold',
    }
});

export default styles;