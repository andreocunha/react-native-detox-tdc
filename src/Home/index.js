import React, { useState } from 'react';
import {
    StatusBar,
    Text,
    View,
    TextInput,
    Button,
} from 'react-native';
import styles from './styles'

export default function Home() {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    function handleSubmit() {
        alert(`Email: ${email} \nPassword: ${password}`);
    }

    return(
        <View style={styles.container}>
            <StatusBar barStyle="dark-content" />
            <Text style={styles.title}>Login</Text>
            <TextInput
                type="text"
                value={email}
                placeholder="Digite o email"
                onChangeText={(text) => setEmail(text)}
                style={styles.input}
            />
            <TextInput
                type="text"
                value={password}
                placeholder="Digite sua senha"
                onChangeText={(text) => setPassword(text)}
                secureTextEntry={true}
                style={styles.input}
            />
            <Button onPress={handleSubmit} title="Entrar" />
        </View>    
    )
}