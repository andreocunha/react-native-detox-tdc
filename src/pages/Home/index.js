import React, { useState } from 'react';
import {
    Alert,
    Text,
    View,
    TextInput,
    TouchableOpacity,
} from 'react-native';
import { Header } from '../../components/Header';
import { isValidEmail, isValidPassword } from '../../utils/loginValidator';
import styles from './styles';

export default function Home() {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    function handleSubmit() {
        if (isValidEmail(email) && isValidPassword(password)) {
            Alert.alert('Login efetuado com sucesso!');
        }
        else {
            Alert.alert('Email ou senha inválidos!');
        }
    }

    return(
        <>
            <Header />
            <View style={styles.container}>
                <Text style={styles.title}>Área de Login</Text>

                <View style={styles.loginArea}>
                    <Text>Email</Text>
                    <TextInput
                        type="text"
                        value={email}
                        placeholder="Digite o email"
                        onChangeText={(text) => setEmail(text)}
                        style={styles.input}
                    />
                    <Text>Senha</Text>
                    <TextInput
                        type="text"
                        value={password}
                        placeholder="Digite sua senha"
                        onChangeText={(text) => setPassword(text)}
                        secureTextEntry={true}
                        style={styles.input}
                    />
                </View>
                
                <TouchableOpacity
                    onPress={handleSubmit}
                    style={styles.button}
                    testID="submitButton"
                >
                    <Text style={styles.textButton}>LOGAR</Text>
                </TouchableOpacity>
            </View>
        </>    
    )
}