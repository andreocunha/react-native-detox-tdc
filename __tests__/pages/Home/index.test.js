import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { Alert } from 'react-native';

import Home from '../../../src/pages/Home';

jest.spyOn(Alert, 'alert');

describe('Home', () => {
    let emailInput;
    let passwordInput;
    let button;
    let json;

    beforeEach(() => {
        const { getByPlaceholderText, getByTestId, toJSON } = render(<Home />);
    
        emailInput = getByPlaceholderText('Digite o email');
        passwordInput = getByPlaceholderText('Digite sua senha');
        button = getByTestId('submitButton');
        json = toJSON();
    });

    afterEach(() => {    
        jest.clearAllMocks();
    });

    it('should render the snapshot', () => {
        expect(json).toMatchSnapshot();
    });

    it('given valid email, valid password and press button, should call alert with success', () => {
        fireEvent.changeText(emailInput, 'valid@email.com');
        fireEvent.changeText(passwordInput, 'Pass123#');
        fireEvent.press(button);

        expect(Alert.alert).toHaveBeenCalledWith('Login efetuado com sucesso!');
        expect(Alert.alert).not.toHaveBeenCalledWith('Email ou senha inv치lidos!');
    });

    it('given valid email, invalid password and press button, should call alert with fail', () => {
        fireEvent.changeText(emailInput, 'valid@email.com');
        fireEvent.changeText(passwordInput, 'invalid');
        fireEvent.press(button);

        expect(Alert.alert).not.toHaveBeenCalledWith('Login efetuado com sucesso!');
        expect(Alert.alert).toHaveBeenCalledWith('Email ou senha inv치lidos!');
    });

    it('given invalid email, valid password and press button, should call alert with fail', () => {
        fireEvent.changeText(emailInput, 'invalid');
        fireEvent.changeText(passwordInput, 'Pass123#');
        fireEvent.press(button);

        expect(Alert.alert).not.toHaveBeenCalledWith('Login efetuado com sucesso!');
        expect(Alert.alert).toHaveBeenCalledWith('Email ou senha inv치lidos!');
    });

    it('given invalid email, invalid password and press button, should call alert with fail', () => {
        fireEvent.changeText(emailInput, 'invalid');
        fireEvent.changeText(passwordInput, 'invalid');
        fireEvent.press(button);

        expect(Alert.alert).not.toHaveBeenCalledWith('Login efetuado com sucesso!');
        expect(Alert.alert).toHaveBeenCalledWith('Email ou senha inv치lidos!');
    });
});