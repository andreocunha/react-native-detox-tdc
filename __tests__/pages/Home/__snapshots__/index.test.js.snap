// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#051933",
        "height": 100,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/logo-alura.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 110,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#051933",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      √Årea de Login
    </Text>
    <View
      style={
        Object {
          "padding": 30,
          "width": "100%",
        }
      }
    >
      <Text>
        Email
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Digite o email"
        style={
          Object {
            "borderColor": "#000",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 44,
            "marginBottom": 20,
            "padding": 8,
            "width": "100%",
          }
        }
        testID="emailInput"
        type="text"
        value=""
      />
      <Text>
        Senha
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Digite sua senha"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#000",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 44,
            "marginBottom": 20,
            "padding": 8,
            "width": "100%",
          }
        }
        testID="passwordInput"
        type="text"
        value=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#051933",
          "borderRadius": 8,
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID="submitButton"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        LOGAR
      </Text>
    </View>
  </View>,
]
`;
